### PR TITLE
PR Labels for CI control 

### PR DESCRIPTION
**Pull request by Fhrozen**
# What

Implement the use of PR labels in the CI tests.

# Why

Reduce the time of the CI by constraining the test to the respective modifications.

### CODE REVIEW

**Comment by Fhrozen** [comment]
@sw005320 @kan-bayashi The labels were implemented in two different configurations:
Using a previous step to skip the evaluations, and probably add values to the matrix strategy, using the labels (such as in `.github/workflows/ci.yaml` jobs: `get_labels`. Or, catching the labels inside the scripts, such as in `ci/install.sh`.
A test was implemented for skipping tests if only docker label is present in the PR, and espnet1 and espnet2 are not present.

Let me know your opinion to modify it.

File: ci/test_integration_espnet1.sh [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+if [ -z "${GITHUB_PR_LABEL_ESPNET1:-}" ]; then
+    log This PR does not involve ESPnet1 fixes. Skipping...
+    exit 0
+fi
+
```
Comment (kan-bayashi) [inline ci/test_integration_espnet1.sh:12]: I think the skip check should be executed in github action since we want to use this in local.

Comment (Fhrozen) [reply ci/test_integration_espnet1.sh:12]: This was a test of using the labels inside the scripts. I will remove then.

Comment (Fhrozen) [reply ci/test_integration_espnet1.sh:12]: I will comment to make as example.

File: ci/install.sh [hunk 2]
```diff
@@ -6,6 +6,19 @@ set -euo pipefail
...
+}
+
+if [ -z "${GITHUB_PR_LABEL_ESPNET1:-}" ] && [ -z "${GITHUB_PR_LABEL_ESPNET2:-}" ]; then
+    # If not Label tag ESPNET 1 or ESPNET 2 but Docker, then skip all tests.
+    if [ -n "${GITHUB_PR_LABEL_DOCKER:-}" ]; then
+        log Only Docker related modifications. Skipping tests.
+        exit 0
+    fi
+fi
+
```
Comment (kan-bayashi) [inline ci/install.sh:21]: Could you remove this one? I think we should manage the condition based on PR-LABEL in github action yaml.

File: ci/test_integration_espnet1.sh [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+# How to use PR Labels:
+# if [ -z "${GITHUB_PR_LABEL_ESPNET1:-}" ]; then
+#     log This PR does not involve ESPnet1 fixes. Skipping...
+#     exit 0
+# fi
+
```
Comment (kan-bayashi) [inline ci/test_integration_espnet1.sh:13]: ditto

File: ci/test_integration_espnet2.sh [hunk 4]
```diff
@@ -14,6 +14,17 @@ gen_dummy_coverage(){
...
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+# How to use PR Labels:
+# if [ -z "${GITHUB_PR_LABEL_ESPNET2:-}" ]; then
+#     log This PR does not involve ESPnet2 fixes. Skipping...
+#     exit 0
+# fi
+
```
Comment (kan-bayashi) [inline ci/test_integration_espnet2.sh:27]: ditto

File: ci/test_python.sh [hunk 5]
```diff
@@ -5,6 +5,11 @@
 
 set -euo pipefail
 
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
```
Comment (kan-bayashi) [inline ci/test_python.sh:12]: Not used. Removed.

**Review by kan-bayashi — CHANGES_REQUESTED** [review]

**Review by kan-bayashi — APPROVED** [review]

**Pull request merged**